src/nilss.jl

```diff
@@ -289,20 +289,27 @@ function store_y_w_vstar!(y, w, vstar, sol, nus, numindvar, numparams, iseg)
   copyto!(_y, (@view sol[1:numindvar,:]))
 
   # fill w
+  # only calculate w one time, w can be reused for each parameter
   for j=1:nus
-    for i=1:numparams
-      indx1 = (j-1)*numindvar*numparams + i*numindvar+1
-      indx2 = (j-1)*numindvar*numparams + (i+1)*numindvar
+    for i=1
+      indx1 = (j-1)*numindvar*1 + i*numindvar+1
+      indx2 = (j-1)*numindvar*1 + (i+1)*numindvar
 
       _w = @view w[i,:,:,iseg, j]
       copyto!(_w, (@view sol[indx1:indx2,:]))
     end
   end
 
+  # copying w for all parameters to use same w
+  for i=2:numparams
+    _w = @view w[i,:,:,iseg,:]
+    copyto!(_w, w[1,:,:,iseg,:])
```

Comment:
w is initialized here: 
https://github.com/SciML/DiffEqSensitivity.jl/blob/749648549bb0068ea5f15952d22e70a9f83f93c0/src/nilss.jl#L148
you can drop the 'numparams' dimension and avoid copying.